### ISSUE TITLE
Menu of StatusNotifier in wrong place with multiple monitors

### ISSUE THREAD

**Issue by Tokariew** [elParaguayo/qtile-extras#88]
Video with the bug: https://youtu.be/jZkR-MZvbZE

I have two monitors, one horizontal, one vertical, when using `StatusNotifier` I get menu in wrong place

```python
notifier_defaults = dict(
    menu_background=colours[0],
    menu_font='FiraCode Nerd Font',
    menu_fontsize=font_size,
    menu_foreground=colours[7],
    opacity=opacity,
    separator_colour=colours[8],
    highlight_colour=colours[12],
)

notifier = widget.StatusNotifier(**notifier_defaults)
```
If I use Mirror on `notifier` widget, then one monitor have menu in correct place, but another not
If i define `StatusNotifier` for each bar then on both monitors i have menu in wrong place

Tested on git version of qtile and qtile-extras

**Comment by elParaguayo** [comment]
Ah, yes. I think I can see the error in the code.

If I submit a patch later today, are you able to test?

Thanks for reporting.

**Comment by Tokariew** [comment]
No problem with it ;)

**Comment by elParaguayo** [comment]
I think #89 should fix this. Can you test and let me know?

**Comment by Tokariew** [comment]
Now… it always display under one of the trays, but when i restart qtile and remove `__pycaches__` It can change under which monitor it display…
`Globalmenu` widget work fine for me, but not `StatusNotifier`. And both have quite similiar `display_menu` function currently

**Comment by elParaguayo** [comment]
Are you using separate instances or a Mirror?

**Comment by elParaguayo** [comment]
Actually, I think I know what the issue is here. It's a bit more complicated. Let me try something hacky first...

**Comment by Tokariew** [comment]
I tried with following widgets in bars:
```

```python
notifier_defaults = dict(
    menu_background=colours[0],
    menu_font='FiraCode Nerd Font',
    menu_fontsize=font_size,
    menu_foreground=colours[7],
    opacity=opacity,
    separator_colour=colours[8],
    highlight_colour=colours[12],
)
notifier = widget.StatusNotifier(**notifier_defaults)

widget_main_list = [
    widget.CurrentLayoutIcon(),
    widget.GroupBox(**groupbox_defaults),
    widget.CurrentScreen(**current_screen_defaults),
    prompt,
    widget.GlobalMenu(**globmenu_defaults),
    widget.TaskList(**tasklist_defaults),
    widget.Mpd2(
        **decorations,
        status_format='{play_status} {artist}/{title}',
        play_states={
            'pause': '', 'play': '喇', 'stop': 'ﭥ'
        },
        max_chars=60,
        idle_format='{idle_message}'
    ),
    widget.WidgetBox(
        widgets=[
            widget.CPU(**decorations, format="龍 {load_percent: >4g}%"),
            widget.Memory(**decorations, format=" {MemPercent: >4g}%"),
            widget.Net(**decorations, format='{down}李{up}'),  # notifier,
        ],
        text_closed="",
        text_open=""
        ),
    widget.StatusNotifier(**notifier_defaults),
    # notifier,
    widget.Clock(format='%Y-%m-%d %H:%M', mouse_callbacks={'Button1': display_cal}),
]
widget_2nd_list = [
    widget.CurrentLayoutIcon(),
    widget.GroupBox(**groupbox_defaults),
    widget.CurrentScreen(**current_screen_defaults),
    prompt,
    widget.GlobalMenu(**globmenu_defaults),
    widget.TaskList(**tasklist_defaults),
    # widget.WidgetBox(widgets=[widget.StatsNotifier(**notifier_defaults)], text_closed="", text_open=""),
    widget.StatusNotifier(**notifier_defaults),
    # notifier,
    # widget.Systray(),
    widget.Clock(format='%Y-%m-%d %a %H:%M')
]
```
And i was switching between version with `notifier` in the code and `widget.StatusNotifier(**notifier_defaults),`

globalmenu work fine if i have seperate instances, so i mainly was testing with seperate instances for notifier too.

**Comment by elParaguayo** [comment]
Yeah, the issue is that the widget wasn't designed to have multiple instances (an oversight by me rather than any specific limitation). I'll need to submit a fix in the main qtile version first and then do an update in qtile-extras.

That may take a bit of time as I'm really busy at work at the moment.

Sorry for the inconvenience.

**Comment by elParaguayo** [comment]
The updated fix in #88 should work once https://github.com/qtile/qtile/pull/3749 is merged.

**Comment by Tokariew** [comment]
Thank you very much. I highly appreciate your work ;)
Installed qtile and qtile_extras with corresponding PR and now it's working fine.
